spring-integration-jmx/src/main/java/org/springframework/integration/jmx/config/MBeanExporterHelper.java

```diff
@@ -92,6 +92,20 @@ public void afterPropertiesSet() throws Exception {
 							&& !(className.endsWith(IntegrationMBeanExporter.class.getName()))){
 						this.siBeansToExcludeWhenNoIMBE.add(beanName);
 					}
+					else if (className.startsWith(SI_ROOT_PACKAGE) // all framework MessageSources
+							&& ((className.endsWith("MessageSource")
+									|| className.endsWith("CorrelatingMessageBarrier")
+									|| className.endsWith("JdbcPollingChannelAdapter")
+									|| className.endsWith("JmsDestinationPollingSource")
+									|| className.endsWith("JpaPollingChannelAdapter")
+									|| className.endsWith("StoredProcPollingChannelAdapter")
+									|| className.endsWith("XmppMessageProducer"))
+									|| className.endsWith("MessageHandler")
+									|| className.endsWith("MessageHandlerChain")
+									|| className.endsWith("BridgeHandler"))) { //TODO: more handlers?
+						// can't think of a better way - any ideas?
```

Comment:
I am trying to maintain backwards compatibility. Now that all components are `@ManagedResource` they will be picked up by a normal context mbean exporter. This code suppresses their inclusion (requiring an IMBE as before).